enhancements/helm3/namespace-scoped-helm-repository-crd.md

```diff
@@ -0,0 +1,158 @@
+
+As an OpenShift developer, I would like to be able to add Helm repositories within my project without asking for an administrator's permissions
+
+### API Extensions
+
+#### Introducing a new namespace-scoped custom resource, ProjectHelmChartRepository
+
+Example definition:
+```yaml
+apiVersion: helm.openshift.io/v1beta1
+kind: ProjectHelmChartRepository
+metadata:
+  name: my-enterprise-chart-repo
+spec:
+  url: https://my.chart-repo.org/stable
+
+  # optional and only needed for UI purposes
+  displayName: myChartRepo
+
+  # optional and only needed for UI purposes
+  description: my private chart repo
+```
+
+To add a new namespace-scoped Helm repository to a desired namespace:
+```shell
+$ cat <<EOF | oc apply --namespace my-namespace -f -
+apiVersion: helm.openshift.io/v1beta1
+kind: ProjectHelmChartRepository
+metadata:
+  name: stable
+spec:
+  url: https://kubernetes-charts-incubator.storage.googleapis.com
+  displayName: Public Helm stable charts
+  description: Public Helm stable charts hosted on HelmHub
+EOF
+
+$ kubectl get projecthelmchartrepositories --namespace my-namespace
+NAME          AGE
+stable        1m
+```
+
+The addition of namespace-scoped Helm repository does not impact behavior of the existing cluster-scoped Helm repository:
+
+```shell
+$ cat <<EOF | oc apply -f -
+apiVersion: helm.openshift.io/v1beta1
+kind: HelmChartRepository
+metadata:
+  name: stable
+spec:
+  url: https://kubernetes-charts.storage.googleapis.com
+  displayName: Public Helm stable charts
+  description: Public Helm stable charts hosted on HelmHub
+EOF
+
+$ cat <<EOF | oc apply --namespace my-namespace -f -
+apiVersion: helm.openshift.io/v1beta1
+kind: ProjectHelmChartRepository
+metadata:
+  name: incubator
+spec:
+  url: https://kubernetes-charts-incubator.storage.googleapis.com
+  displayName: Public Helm charts in incubator state
+EOF
+
+$ kubectl get helmchartrepositories
+NAME          AGE
+cluster       3h30m
+stable        1m
+
+$ kubectl get projecthelmchartrepositories --namespace my-namespace
+NAME          AGE
+incubator     1m
+```
+
+#### How would the UI render the namespace-scoped Helm repository
+
+As defined in [previous helm enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/helm3/console.md#how-would-the-ui-discover-the-charts), the UI will invoke the `/api/helm/charts/index.yaml` endpoint and the endpoint will proxy requests to the configured chart repositories.
+
+Then the chart repository proxy will use all configured chart repositories, including cluster-scoped and available namespace-scoped Helm repositories in the current namespace, and deliver to the UI an aggregated index file. The UI then renders the developer catalog with the aggregated index file.
+
+### Risks and Mitigations
+
+## Design Details
+
+### Test Plan
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
```

Comment:
presumably there's nothing that needs to be done here, but you should at least "N/A" the section to show us you've thought about it (same for all the other sections you left blank).

and there is at least the slight implication that if they downgrade, the namespaced instances will still exist, but nothing will be consuming them, presumably.

Reply:
Thanks for the review! That is correct, if a user downgrades then the instances are not affected but the console will not be consuming/rendering them. Will update this and other blank fields.